test_unstructured_ingest/test-ingest-elasticsearch.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+cd "$SCRIPT_DIR"/.. || exit 1
+
+# Create the elasticsearch cluster and get the container id
+output=$(docker run -d --rm -p 9200:9200 -p 9300:9300 -e "xpack.security.enabled=false" -e "discovery.type=single-node" docker.elastic.co/elasticsearch/elasticsearch:8.7.0)
+container_id=$(echo "$output" | cut -c 1-12)
+
+url="http://localhost:9200/_cluster/health"
+status_code=0
+retry_count=0
+max_retries=6
+
+# Check the cluster status repeatedly until it becomes live or maximum retries are reached
+while [ "$status_code" -ne 200 ] && [ "$retry_count" -lt "$max_retries" ]; do
+
+  # Send a GET request to the cluster health API
+  response=$(curl -s -o /dev/null -w "%{http_code}" "$url")
+  status_code="$response"
+
+  # Process the files only when the ES cluster is live
+  if [ "$status_code" -eq 200 ]; then
+    echo "Cluster is live."
+
+    python examples/ingest/elasticsearch/elasticsearch_cluster.py
```

Comment:
it's kind of awkward that we rely on a module that lives in `examples/` in order to run a test.
that said, I don't have great alternatives. I think the best I can think of is: create a new folder in `scripts/`, that folder should contain the shell script that launches / waits / retries (that I mentioned in the comment above this), `es_cluster_config.py`, and `elasticsearch_cluster.py`. then both the examples/../ingest.sh and `test-ingest-elasticsearch.sh` can simply call that one script in one line and afterwards have a running instance to test against.

Reply:
Changes applied! (all isolated code moved into `scripts` folder, both  `examples/../ingest.sh` and `test-ingest-elasticsearch.sh` calling the isolated code from there. shellcheck ignored for those particular script calls since it gave a SC1091 warning.)